engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Vector.java

```diff
@@ -38,23 +42,31 @@ private Vector(Object storage) {
     this.storage = storage;
   }
 
+  public static InvokeFunctionNode createValueForIndexNode() {
+    return InvokeFunctionNode.build(
+        new CallArgumentInfo[] {new CallArgumentInfo()},
+        InvokeCallableNode.DefaultsExecutionMode.EXECUTE,
+        InvokeCallableNode.ArgumentsExecutionMode.EXECUTE);
```

Comment:
You can probably experiment with removing arguments execution mode. It's a leftover from a bygone era and the only thing it potentially does is a tiny performance improvement. But I'm not sure it does, so better check.